### DISCUSSION TITLE
Why are nextcord views so complicated to make and use?

### DISCUSSION THREAD

**Discussion by Tom-Sumner** [nextcord/nextcord#437] [Q&A] [answered]
before I switched I used discord-interactions which had the same features but was so much less complicated and wasn't using classes or anything. why has nextcord made it difficult for devs to use selects and buttons  compared to much smaller libraries.

Nextcord example:
```
class HelpDropdown(nextcord.ui.Select):
	def __init__(self, command):
		pass
		super().__init__(placeholder="Choose a command", min_values=1, max_values=1, options=[SelectOption(label="Hey", value="Hello")])

	async def callback(self, ctx: nextcord.Interaction):
	    await ctx.response.send_message(f"You clicked {self.values[0]}")


class HelpView(nextcord.ui.View):
	def __init__(self, command, *, timeout: Optional[float] = 120.0):
		super().__init__(timeout=timeout)
		self.add_item(HelpDropdown(command))
```
Example of discord-interaction select menu
```
await ctx.send("This message has a dropdown", components=[Select(placeholder="Select your job", options=[SelectOptions(name="op1", value="op1")], custom_id="123")])
```

**Comment by DenverCoder1** [comment, marked as answer]
This was not a decision by Nextcord, but rather Discord.py 2.0 added Views which is why every fork of dpy has it implemented like this. I wouldn't be against adding an alternate method of doing it, but as the functionality already exists it isn't a high priority. It's possible in Nextcord v3 it will be implemented differently, but you'd have to ask someone like @TAG-Epic about that.

**Comment by XenonChek** [comment]
I think it gives more examples to understand it.
